linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -828,6 +828,15 @@ object Emitter {
               StringArgConstructorName)
         },
 
+        // See systemIdentityHashCode in CoreJSLib
+        callMethod(BoxedBooleanClass, hashCodeMethodName),
+        callMethod(BoxedDoubleClass, hashCodeMethodName),
+        callMethod(BoxedStringClass, hashCodeMethodName),
+        callMethod(BoxedUnitClass, hashCodeMethodName),
+        cond(config.esFeatures.allowBigIntsForLongs) {
+          callMethod(BoxedLongClass, hashCodeMethodName)
```

Comment:
Hmmm... looking at the implementation of `SystemIdentityHashCode`, it seems it will cause an infinite loop if:
- `!useBigIntsForLong`
- The hash of a `BigInt` is taken.

(this is unrelated to this PR).

Reply:
Yes, I think `SystemIdentityHashCode` is broken in some way. I'll investigate that after this PR.